src/ducks/context/SelectionContext.jsx

```diff
@@ -15,6 +15,31 @@ export const useSelectionContext = () => {
   return useContext(SelectionContext)
 }
 
+export const useSelectionToggler = item => {
+  const {
+    addToSelection,
+    isSelected,
+    removeFromSelection,
+    isSelectionModeEnabled
+  } = useSelectionContext()
+
+  const isItemSelected = useMemo(() => isSelected(item), [isSelected, item])
```

Comment:
Since itemSelected is a boolean value, triple equality will work even if it's a "new" value. Is isSelected really expensive ?

Reply:
tu veux dire qu'un `const isItemSelected === isSelected(item)` suffit ? Il ne va pas être recréé à chaque render ?

Reply:
Oui je veux dire ça.

Qu'est ce qui va être recréé ? Si la valeur du booléen renvoyé par isSelected est la même entre deux renders, comme c'est un booléen, true === true même si c'est un nouveau true.

Reply:
heu en fait c'est plutôt `const isItemSelected = isSelected(item)`...

désolé je comprends pas pourquoi `const isItemSelected = isSelected(item)` est équivalent à  `const isItemSelected = useMemo(() => isSelected(item), [isSelected, item])` 

Dans le premier cas `isItemSelected` ne va jamais être caché et `isSelected(item)` va être exécuté à chaque render, contrairement au deuxième cas qui n'exécute `isSelected(item)` que sur changement des 2 dépendances, non ?

Reply:
ah tu ferais un `const isItemSelected = isSelected(item) === true` ??

Reply:
Ok tu parlais bien de `const isItemSelected = isSelected(item)` mais de la triple égalité dans le receveur de `isItemSelected` 👍